File: AspectJCacheExample/src/main/java/com/blogspot/miguelinlas3/aspectj/transaction/manager/MockTransactionManager.java
package com.blogspot.miguelinlas3.aspectj.transaction.manager;

/**
 * Stupid class to emulate the basic operations of a transaction manager
 * 
 * @author migue
 * 
 */
public class MockTransactionManager {

	public void openTransaction(){
		System.out.println("Opening transaction");
	}
	
	public void closeTransaction(){
		System.out.println("Closing transaction");
	}
	
	public void rollback(){
		System.out.println("Roll back previous work");
	}
	
	public void commit(){
		System.out.println("Commiting the work");
	}
}

File: scala-talk/src/main/scala/com/blogspot/miguelinlas3/scala/intro/highorderunction/transaction/manager/MockTransactionManager.scala
package com.blogspot.miguelinlas3.scala.intro.highorderunction.transaction.manager

class MockTransactionManager {

	def openTransaction(): Unit = {
		println ("Opening transaction")
	}
	
	def closeTransaction(): Unit = {
		println("Closing transaction");
	}
	
	def rollback(): Unit = {
		println("Roll back previous work");
	}
	
	def commit(): Unit = {
		println("Commiting the work");
	}
}